plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -355,9 +390,19 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
 		if !hasUsageStats(&resourceQuota) {
 			return nil, admission.NewForbidden(a, fmt.Errorf("status unknown for quota: %s", resourceQuota.Name))
 		}
+		requiredResourcesSet := quota.ToSet(requiredResources)
+		mustHaveCoveringQuotaSet = mustHaveCoveringQuotaSet.Difference(requiredResourcesSet)
```

Comment:
Ok, its because you've changed the meaning of this variable.  Before, it was the list of quotaed resources that must be satisfied and you've changed it here to the list of unsatisfied-but-required-quotaed-resources.

I think the code would read cleanest if you track all the `requiredResources` from the entire loop and then do a `HasAll` (or `Difference` with a good comment to avoid the conversion).